OpenRA.Mods.Common/Warheads/SpreadDamageWarhead.cs

```diff
@@ -58,7 +59,7 @@ public override void DoImpact(WPos pos, Actor firedBy, IEnumerable<int> damageMo
 
 			// This only finds actors where the center is within the search radius,
 			// so we need to search beyond the maximum spread to account for actors with large health radius
-			var hitActors = world.FindActorsInCircle(pos, Range[Range.Length - 1] + TargetExtraSearchRadius);
+			var hitActors = world.FindActorsInCircle(pos, Range[Range.Length - 1] + TargetExtraSearchRadius).Where(a => a.Info.HasTraitInfo<HealthInfo>());
```

Comment:
This introduces a second trait(info) lookup per actor.  It would be better to keep the one query in the loop, but change the conditional to `if (healthInfo == null) continue;`